File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feature_workout/data/model/FastActivityForm.kt
package com.example.happyhabits.feature_application.feature_workout.data.model

import kotlinx.serialization.Serializable

@Serializable
data class FastActivityForm(
    val userId: String,
    val date: String,
    val type: String,
    val time: String,
    val duration: String,
    val notes: String? = "",
    val quantity: Float,
    val elevation: Float
)

File: API-Server/Happy Habits App/Forms/FastActivityForm.cs
using MongoDB.Libmongocrypt;
using System.Text.Json.Serialization;

namespace Happy_Habits_App.Forms
{
    public class FastActivityForm : HabitForm
    {
        [JsonPropertyName("type")]
        public string? Type { get; set; } = null;
        [JsonPropertyName("time")]
        public string? Time { get; set; } = null;
        [JsonPropertyName("duration")]
        public string? Duration { get; set; } = null;
        [JsonPropertyName("notes")]
        public string? Notes { get; set; } = null;
        [JsonPropertyName("quantity")]
        public float Quantity { get; set; }
        [JsonPropertyName("elevation")]
        public float Elevation { get; set; }

        public bool IsValid
        {
            get
            {
                return !string.IsNullOrEmpty(UserId) &&
                       !string.IsNullOrEmpty(Date) &&
                       !string.IsNullOrEmpty(Time) &&
                       !string.IsNullOrEmpty(Duration);
            }
        }
    }
}
